### PR TITLE
DROOLS-3225: [DMN Designer] Decision table output data type is not stored

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-3225

### CODE REVIEW

**Comment by karreiro** [comment]
Jenkins please retest this.

**Comment by jomarko** [comment]
@manstis can not approve, see my [comment](https://issues.jboss.org/browse/DROOLS-3225?focusedCommentId=13655586&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13655586), no comments to code.

**Comment by manstis** [comment]
Jenkins please retest this.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/DecisionTableGrid.java [hunk 1]
```diff
@@ -225,15 +229,27 @@ private OutputClauseColumn makeOutputClauseColumn(final OutputClause oc) {
...
+                        metaData.add(new OutputClauseColumnExpressionNameHeaderMetaData(hasExpression,
+                                                                                        expression,
+                                                                                        hasName,
+                                                                                        clearDisplayNameConsumer(true),
+                                                                                        setDisplayNameConsumer(true),
+                                                                                        setTypeRefConsumer(),
+                                                                                        cellEditorControls,
+                                                                                        headerEditor,
+                                                                                        Optional.of(translationService.getTranslation(DMNEditorConstants.DecisionTableEditor_EditOutputClause))));
+                    }
```
Comment (karreiro) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/DecisionTableGrid.java]: I would DRY it a little bit with a method like this:
```java
private BiConsumer<HasTypeRef, QName> setTypeRefConsumer(final OutputClause oc,
                                                         final DecisionTable dtable) {
    if (dtable.getOutput().size() == 1) {
        return setTypeRefOnHasExpressionAndOutputClauseConsumer(oc);
    }
    return setTypeRefConsumer();
}
```
..but feel free to keep it as is. It's just a method call.. ;-)

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/DecisionTableGrid.java]: Updated. Thanks for the suggestion.

**Pull request merged**